src/controllers/requestController.js

```diff
@@ -1,67 +1,119 @@
+/* eslint-disable require-jsdoc */
+/* eslint-disable no-underscore-dangle */
 import model from '../database/models';
 import { findRooms } from '../services/roomService';
+import requestService from '../services/requestService';
 
-//creating a request
-const createRequest = async (req, res) => {
-  try {
-    const { idUser, idRoom, dateStart, dateEnd } = req.body,
-      { id } = req.userData;
-    const existingRoom = await findRooms(idRoom);
-
-    if(!existingRoom) return res.status(400).json({ message: 'Room id does not exist' });
-  
-    const newRequest = {
-      idUser: req.userData.id,
-      idRoom,
-      dateStart,
-      dateEnd
-    };
-    const request = await model.request.create(newRequest);
-
-    return res.status(201).json({ message: 'Request created successfully!', request});
-  } catch (error) {
-    return res.status(500).json({error: "Internal server error!"});
+const { request } = model,
+  { findRequestById, findRequestByUserId, findRequestByRoomId } = requestService;
+
+export default class requestController {
+// getting all requests
+  static async getAllRequests(req, res) {
+    const userId = req.userData.id,
+      existingRequests = await findRequestByUserId(userId);
+    if (existingRequests.length === 0) return res.status(404).json({ message: res.__('Request does not exist!') });
+
+    return res.status(200).json({ message: res.__('Requests found successfully!'), existingRequests });
   }
-};
-
-//updating a request
-const updateRequest = async (req, res) => {
-  try {
-    const { idRequest } = req.params;
-    const [ updated ] = await model.request.update(req.body, {
-      where: { id: idRequest }
-    });
-    if (updated) {
-      const updatedRequest = await model.request.findOne({ where: { id: idRequest } });
-      return res.status(200).json({ request: updatedRequest });
+
+  // get one request
+  static async getOneRequest(req, res) {
+    try {
+      const userId = req.userData.id,
```

Comment:
Define common code blocks as a class/static method that you just call where you need, say userRequestIds